widget/textgrid.go

```diff
@@ -292,15 +295,27 @@ func (t *TextGrid) SetStyleRange(startRow, startCol, endRow, endCol int, style T
 // CreateRenderer is a private method to Fyne which links this widget to its renderer
 func (t *TextGrid) CreateRenderer() fyne.WidgetRenderer {
 	t.ExtendBaseWidget(t)
-	render := &textGridRenderer{text: t}
-	render.updateCellSize()
 
 	th := t.Theme()
 	v := fyne.CurrentApp().Settings().ThemeVariant()
 	TextGridStyleDefault = &CustomTextGridStyle{}
 	TextGridStyleWhitespace = &CustomTextGridStyle{FGColor: th.Color(theme.ColorNameDisabled, v)}
 
-	return render
+	var scroll *widget.Scroll
+	content := newTextGridContent(t)
+	objs := make([]fyne.CanvasObject, 1)
+	if t.Scroll == widget.ScrollNone {
+		scroll = widget.NewScroll(nil)
+		objs[0] = content
+	} else {
+		scroll = widget.NewScroll(content)
+		scroll.Direction = t.Scroll
+		objs[0] = scroll
+	}
+	t.scroll = scroll
+	r := &textGridRenderer{text: content, scroll: scroll}
+	r.SetObjects(objs)
```

Comment:
This can be cleaned up a bit:
```suggestion
	var inScroller fyne.CanvasObject
	content := newTextGridContent(t)
	objs := []fyne.CanvasObject{content}
	if t.Scroll != widget.ScrollNone {
		inScroller = content
		scroll.Direction = t.Scroll
		objs[0] = scroll
	}
	t.scroll = widget.NewScroll(inScroller)
	r := &textGridRenderer{text: content, scroll: t.scroll}
	r.SetObjects(objs)
```

Reply:
This feels subjective - can it be definitive that creating "inScroller" to avoid an else is better?

Reply:
Hmm, yeah. Might be subjective. I just try to avoid branches as much as possible because it is less code indented to the right. Do as you wish :)

Reply:
I see that but it is not removing an indent (as that level deep was already there).
I was a little confused because the nil content of Scroller becomes implicit which could look like a bug.